### DISCUSSION TITLE
exclude subpath from kustomize.toolkit.fluxcd.io/v1beta2

### DISCUSSION THREAD

**Discussion by ahgraber** [fluxcd/flux2#2599] [Q&A] [answered]
Simplistically, my repo is set up as below.  
All of my applications are in an `apps` folder that are managed by a global kustomize.toolkit.fluxcd.io/v1beta2 for path ./apps.
However, `./apps/app3` has a dependency on `./apps/app2`. 
Is there a way to exclude `app3` from the global `./apps` kustomize so I can set up a kustomize.toolkit.fluxcd.io/v1beta2 for `app3` with a health assessment against `app2`?
Or do I need to set up a more granular structure with a kustomize.toolkit.fluxcd.io/v1beta2 per app within `./apps`?

I hope that makes sense.

```txt
.
├── base
│   ├── flux-system 
│   │   └── charts         
│   │       ├── ...
│   ├── apps.yaml                      # kustomize.toolkit.fluxcd.io/v1beta2 for path ./apps
│   └── core.yaml                      # kustomize.toolkit.fluxcd.io/v1beta2 for path ./core
├── apps
│   ├── app1
│   ├── app2
│   ├── app3                           # dependency on app2
│   └── app4
├── core
└── README.md
```

**Comment by stefanprodan** [comment, marked as answer]
You can create a dedicated GitRepository for it and use the `spec.ignore` to exclude a path, docs here: https://fluxcd.io/docs/components/source/gitrepositories/#excluding-files. While this will work, it feels like a hack to me, I would create a Flux Kustomization per app, that's how we designed Flux to work.

**Reply by ahgraber** [reply]
Thanks!
